diff --git a/readpass.c b/readpass.c
index d42b1185d..d0ff2c0b0 100644
--- a/readpass.c
+++ b/readpass.c
@@ -250,6 +250,11 @@ notify_start(int force_askpass, const char *fmt, ...)
 	xvasprintf(&prompt, fmt, args);
 	va_end(args);
 
+	force_askpass = force_askpass ||
+	                ((s = getenv(SSH_ASKPASS_REQUIRE_ENV)) != NULL &&
+	                 ((strcmp(s, "force") == 0) ||
+	                   strcmp(s, "notify-force") == 0));
+
 	if (fflush(NULL) != 0)
 		error_f("fflush: %s", strerror(errno));
 	if (!force_askpass && isatty(STDERR_FILENO)) {
@@ -263,8 +268,7 @@ notify_start(int force_askpass, const char *fmt, ...)
 		goto out;
 	}
 	if (getenv("DISPLAY") == NULL && getenv("WAYLAND_DISPLAY") == NULL &&
-	    ((s = getenv(SSH_ASKPASS_REQUIRE_ENV)) == NULL ||
-	    strcmp(s, "force") != 0)) {
+	    !force_askpass) {
 		debug3_f("cannot notify: no display");
 		goto out;
 	}
